frontend/ui/screensaver.lua

```diff
@@ -23,18 +23,41 @@ local _ = require("gettext")
 local Screen = Device.screen
 local T = require("ffi/util").template
 
-local screensaver_provider = {
-    ["jpg"] = true,
-    ["jpeg"] = true,
-    ["png"] = true,
-    ["gif"] = true,
-    ["tif"] = true,
-    ["tiff"] = true,
+-- Migrate old settings
+if G_reader_settings:readSetting("screensaver_type", "disable") == "message" then
+    G_reader_settings:saveSetting("screensaver_type", "disable")
+    G_reader_settings:makeTrue("screensaver_show_message")
+end
+if G_reader_settings:has("screensaver_no_background") then
+    if G_reader_settings:isTrue("screensaver_no_background") then
+        G_reader_settings:saveSetting("screensaver_background", "none")
+    end
+    G_reader_settings:delSetting("screensaver_no_background")
+end
+if G_reader_settings:has("screensaver_white_background") then
+    if G_reader_settings:isTrue("screensaver_white_background") then
+        G_reader_settings:saveSetting("screensaver_background", "white")
+    end
+    G_reader_settings:delSetting("screensaver_white_background")
+end
+
+local Screensaver = {
+    screensaver_provider = {
+        ["jpg"] = true,
```

Comment:
or just
```suggestion
        jpg = true,
```

Reply:
Btw, what's this list of filetypes about exactly? Not really related to this PR of course.

Reply:
It's the filter for choosing image files manually.